plenum/server/consensus/ordering_service_msg_validator.py

```diff
@@ -0,0 +1,109 @@
+from typing import Tuple, Optional
+
+from common.exceptions import LogicError
+from plenum.common.messages.internal_messages import NewViewCheckpointsApplied
+from plenum.common.messages.node_messages import PrePrepare, Commit, Prepare
+from plenum.common.stashing_router import DISCARD, PROCESS
+from plenum.common.types import f
+from plenum.common.util import compare_3PC_keys
+from plenum.server.consensus.consensus_shared_data import ConsensusSharedData
+from plenum.server.replica_validator_enums import STASH_WAITING_NEW_VIEW, STASH_WATERMARKS, STASH_VIEW, STASH_CATCH_UP, \
+    ALREADY_ORDERED, OUTSIDE_WATERMARKS, CATCHING_UP, FUTURE_VIEW, OLD_VIEW, WAITING_FOR_NEW_VIEW
+
+
+class OrderingServiceMsgValidator:
+
+    def __init__(self, data: ConsensusSharedData):
+        self._data = data
+
+    # TODO: this method is created for compatibility adn tests only
+    # validate_XXX methods will be used later
+    def validate(self, msg) -> Tuple[int, Optional[str]]:
+        if isinstance(msg, PrePrepare):
+            return self.validate_pre_prepare(msg)
+        if isinstance(msg, Prepare):
+            return self.validate_prepare(msg)
+        if isinstance(msg, Commit):
+            return self.validate_commit(msg)
+        if isinstance(msg, NewViewCheckpointsApplied):
+            return self.validate_new_view(msg)
+        raise LogicError("Unknown message type")
+
+    def validate_pre_prepare(self, msg: PrePrepare) -> Tuple[int, Optional[str]]:
+        # we discard already ordered PrePrepares
+        # check for discarded first of all
+        view_no = getattr(msg, f.VIEW_NO.nm, None)
+        pp_seq_no = getattr(msg, f.PP_SEQ_NO.nm, None)
+        if self.has_already_ordered(view_no, pp_seq_no):
+            return DISCARD, ALREADY_ORDERED
+
+        return self._validate_3pc(msg)
+
+    def validate_prepare(self, msg: Prepare) -> Tuple[int, Optional[str]]:
+        # process Prepares that have been already ordered
+        # to re-order batches from NEW_VIEW msg
+        # prepares below low watermark have been discarded anyway
+        return self._validate_3pc(msg)
```

Comment:
Not a call to action, just a rant - the more I look at separate validator thing the more I feel like that we should not separate validator methods into separate files - they are not THAT big, they are not actually shared between services and this separation just requires more chasing through source code when looking at processing entry points.